src/DIRAC/WorkloadManagementSystem/Service/VirtualMachineManagerHandler.py

```diff
@@ -0,0 +1,558 @@
+      siteList = None
+    return self.getCEInstances(siteList=siteList, ceList=ceList, vo=vo)
+
+  types_stopInstance = [six.string_types, six.string_types, six.string_types]
+
+  def export_stopInstance(self, site, endpoint, nodeID):
+
+    return self.stopInstance(site, endpoint, nodeID)
+
+  types_getPilotOutput = [six.string_types]
+
+  def export_getPilotOutput(self, pilotReference):
+
+    return self.getPilotOutput(pilotReference)
+
+  types_checkVmWebOperation = [six.string_types]
+
+  def export_checkVmWebOperation(self, operation):
+    """
+    return true if rpc has OPERATOR
+    """
+    if OPERATOR in self.rpcProperties:
+      return S_OK('Auth')
+    return S_OK('Unauth')
+
+  types_insertInstance = [six.string_types, six.string_types, six.string_types, six.string_types, six.string_types]
+
+  def export_insertInstance(self, uniqueID, imageName, instanceName, endpoint, runningPodName):
+    """
+    Check Status of a given image
+    Will insert a new Instance in the DB
+    """
+    res = self.virtualMachineDB.insertInstance(uniqueID, imageName, instanceName, endpoint, runningPodName)
+    self.__logResult('insertInstance', res)
+
+    return res
+
+  types_getUniqueID = [six.string_types]
+
+  def export_getUniqueID(self, instanceID):
+    """
+    return cloud manager uniqueID from VMDIRAC instanceID
+    """
+    res = self.virtualMachineDB.getUniqueID(instanceID)
+    self.__logResult('getUniqueID', res)
+
+    return res
+
+  types_getUniqueIDByName = [six.string_types]
+
+  def export_getUniqueIDByName(self, instanceName):
+    """
+    return cloud manager uniqueID from VMDIRAC name
+    """
+    result = self.virtualMachineDB.getUniqueIDByName(instanceName)
+    self.__logResult('getUniqueIDByName', result)
+
+    return result
+
+  types_setInstanceUniqueID = [six.integer_types, six.string_types]
+
+  def export_setInstanceUniqueID(self, instanceID, uniqueID):
+    """
+    Check Status of a given image
+    Will insert a new Instance in the DB
+    """
+    res = self.virtualMachineDB.setInstanceUniqueID(instanceID, uniqueID)
+    self.__logResult('setInstanceUniqueID', res)
+
+    return res
+
+  types_declareInstanceSubmitted = [six.string_types]
+
+  def export_declareInstanceSubmitted(self, uniqueID):
+    """
+    After submission of the instance the Director should declare the new Status
+    """
+    res = self.virtualMachineDB.declareInstanceSubmitted(uniqueID)
+    self.__logResult('declareInstanceSubmitted', res)
+
+    return res
+
+  types_declareInstanceRunning = [six.string_types, six.string_types]
+
+  def export_declareInstanceRunning(self, uniqueID, privateIP):
+    """
+    Declares an instance Running and sets its associated info (uniqueID, publicIP, privateIP)
+    Returns S_ERROR if:
+    - instanceName does not have a "Submitted" entry
+    - uniqueID is not unique
+    """
+    gLogger.info('Declare instance Running uniqueID: %s' % (uniqueID))
+    if VM_RPC_OPERATION not in self.rpcProperties:
```

Comment:
No this should not be here, but in the configuration.

Reply:
These are now done in the config.